frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-manager.ts

```diff
@@ -137,6 +137,11 @@ export class OlMapManager {
             .select(selectRestrictedViewport)
             .pipe(takeUntil(this.destroy$))
             .subscribe((viewport) => {
+                if (viewport?.viewportType === 'eoc') {
+                    // Do not restrict the view for EOCs, since it's not a map (no size)
+                    return;
+                }
```

Comment:
Wouldn't it make sense to combine this with the `if (!viewport)` a couple of lines later?

Or, if the map is not in the DOM (see previous comment), you should be able to completely remove this additional check